ceph.spec.in

```diff
@@ -1524,6 +1525,9 @@ cmake .. \
     -DWITH_JAEGER:BOOL=OFF \
 %endif
     -DWITH_GRAFANA:BOOL=ON \
+%if %{with sccache}
+    -DWITH_SCCACHE=ON \
```

Comment:
wont we need to add in `BuildRequires` `sccache` as well to ensure that its installed and available?

Reply:
yeah, and we might want to guard it with `%{with sccache}` as well.

Reply:
This is a good point. But can we express `BuildRequires` for something that isn't packaged?

This isn't meant to be used for release builds; mainly to _allow_ using the tool to speed up CI builds.

Reply:
we do it for `make`/`Ninja` etc ref: https://github.com/ceph/ceph/pull/40097/files - so we should do it for this as well since this is a build dependency IF `sccache` is turn on

i would say it should be there... is `with_sccache` is defined. And we can define it ONLY for CI builds if we dont want it for release builds. Default behavior is not changing..

Reply:
@zmc has a point as (without testing to verify it) I don't think that BuildRequires considers unpackaged binaries in the system - I do belive it uses the rpm package db to resolve package names and binaries. If you still disagree we can test it to make sure.

@zmc one way to satisfy the request would be to use a packaged version of sccache. Unfortunately, it doesn't appear packaged for either fedora or centos. At least one person created a COPR (see https://copr.fedorainfracloud.org/coprs/alebastr/sccache/ ). This COPR is a bit out of date and probably is not something we ought to use directly. However, we could take inspiration from this work and establish our own COPR build. WDYT?

Reply:
I suggested possibly a COPR in private conversation with Zack.  It's not a bad option.  However, I don't think BuildRequires is appropriate or necessary; this is an optional build optimization that requires more setup than just finding the software, anyway, so pointing toward upstream instructions on how to install is fine.

Reply:
The upstream project does publish binary artifacts for the platforms we care about. _If_ sccache usage in our community starts to really take off, let's revisit the idea of maintaining packages and adding the `BuildRequires`.

Reply:
Agree, with one small suggestion: rename the option something like "unmanaged_sccace" or "external_sccache" with a short comment capturing the discussion. The name will hint to people that the binary we're using is a bit special and the comment will elaborate a bit on that.

Reply:
I think the option rename would be clunky/confusing.  I agree it would be nice to note somewhere that you must ensure sccache is available to use this option.  It sure would be nice if bcond_with had some way to add help text.  Failing that, yeah, I think a comment in the RPM is sufficient, just noting "if you use this option, you need to make sure sccache is available".

Reply:
We don't seem to have any precedent with adding things like error messaging directly to the spec file, but [I did add some to CMakeLists.txt](https://github.com/ceph/ceph/blob/main/CMakeLists.txt#L86-L90) a while back.

Is "Can't find sccache. Is it installed?" sufficient or would we want more?

Reply:
satisfies me.  rpmbuilders-from-spec-file can figure it out.